### PR TITLE
Respect output flag when building dylibs

### PR DESCRIPTION
**Pull request by isaacvando**

### CODE REVIEW

File: crates/compiler/roc_target/src/lib.rs [hunk 1]
```diff
@@ -9,7 +9,8 @@ use target_lexicon::Triple;
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum OperatingSystem {
     Windows,
-    Unix,
+    Linux,
+    MacOS,
     Wasi,
 }
```
Comment (isaacvando) [inline crates/compiler/roc_target/src/lib.rs:15]: I'm not completely sold on this solution so I am open to doing something else.

Comment (isaacvando) [reply crates/compiler/roc_target/src/lib.rs:15]: Alternatively I could pass a boolean into `link_macos` and `link_linux` that determines whether or not the extension should be overridden. But that seems like a bandaid.

**Comment by isaacvando** [comment]
I will look later to figure out what I did to break all the tests 😅

**Comment by Anton-4** [comment]
Feel free to ask me for review when the tests are fixed @isaacvando :)

**Pull request closed without merging**